### PR TITLE
Feature UI

### PR DESCRIPTION
**Pull request by malceore**
Opening this Pr to merge in several changes I made over T-day break. Check them out locally and make sure they work so I can merge them.

### CODE REVIEW

File: js/combat.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+function updateHealth(menu){
```
Comment (malceore) [inline js/combat.js:1]: regrouped the combat stuff into a separate file to keep index.js more about state management and higher functions.

File: js/map.js [hunk 2]
```diff
@@ -41,21 +41,9 @@ function loadMapCollisions(image, width, height){
     return pix;
```
Comment (malceore) [inline js/map.js:41]: Deletes in these files are mostly clean up of unused functions.

File: js/ui.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+"use strict"
```
Comment (malceore) [inline js/ui.js:1]: Start of the new UI class for some reusable graphical functions like buttons that are made consistent by these font and color settings at the top.

**Review by IJMadigan — COMMENTED** [review]
Looks good

**Review by HelmetBugg — COMMENTED** [review]
Great work bossman, approved.

**Review by HelmetBugg — APPROVED** [review]

**Review by IJMadigan — APPROVED** [review]
Looking good

**Comment by malceore** [comment]
[Refactor menus under a UI class.](https://trello.com/c/EFGV96H6/74-refactor-menus-under-a-ui-class)

**Pull request merged**